packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +240,135 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def query_dns_server(fqdn, zone, local_ip)
+      query_dns = Resolv::DNS.new(nameserver: [local_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      existing_ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      existing_serial = soa_record.serial if soa_record
+      new_serial = (existing_serial + 1).to_s if existing_serial
+      {
+        existing_ip: existing_ip,
+        new_serial: new_serial
+      }
+    end
+
+    def parse_zone_file(zone_file)
+      run_cmd('rndc freeze')
+      zone_file_data = File.read(zone_file).split("\n")
+      # find the character A enclosed by any whitespace chars
+      a_record = zone_file_data.find { |line| !(line =~ /\sA\s/).nil? }
+      existing_ip = a_record.split('A').last.strip if a_record
+      serial_line = zone_file_data.find { |line| line.downcase.include?('serial') }
+      existing_serial = serial_line.split('Serial').first.to_i if serial_line
+      new_serial = (existing_serial + 1).to_s if existing_serial
+      {
+        existing_ip: existing_ip,
+        new_serial: new_serial
+      }
+    end
+
+    def get_dns_data
+      STDOUT.puts 'gathering DNS data...'
+      local_ip = '127.0.0.1'
+      zone = @scenario_answers['foreman_proxy']['dns_zone']
+      reverse_zone = @scenario_answers['foreman_proxy']['dns_reverse']
+      if reverse_zone.is_a?(Array)
+        reverse_zone = reverse_zone.first
+      end
+      old_fqdn = @old_hostname
+      soa_admin_domain = "root.#{zone}"
+      new_fqdn = @new_hostname
+      key_file = @scenario_answers['foreman_proxy']['keyfile']
+      begin
+        STDOUT.puts 'querying local DNS server...'
+        ip_serial_data = query_dns_server(old_fqdn, zone, local_ip)
+        existing_ip = ip_serial_data[:existing_ip]
+        new_serial = ip_serial_data[:new_serial]
```

Comment:
seems like right around here is where we should create the new serial based on the old one

Reply:
@jturel I started to make this change but realized I feel like it doesn't make as much sense as one would think.  I have to create both `serial` and `reverse_serial` values for both `query_dns_server` and `parse_zone_file` methods, and moving that incrementation to `get_dns_data` creates a lot of repeated code.

Reply:
specifically, instead of getting the incremented value directly from the method I would have to return the unincremented value, then increment it in 2 places in `get_dns_data`.  It would add 2 lines to `get_dns_data` and subtract 1 line from each of the other methods... doesn't seem worth it to me.

Reply:
Having a clear separation of responsibility is more important than saving a few lines of code. `get_dns_data` is the method concerned with determining the future state of the DNS so it should know/care about incrementing the serial. If we wanted to change how we generate the new serial then we'd have to update two methods - that's an increased risk of breaking something and the code is still duplicated in two places.

Reply:
@jturel Hadn't thought of it that way, you're right.  Updated.